src/components/NftRewards/NftRewardsSection.tsx

```diff
@@ -117,14 +149,33 @@ export function NftRewardsSection() {
     // set selected as highest reward tier above a certain amount
     if (highestEligibleRewardTier) {
       selectTier(rewardTiers.indexOf(highestEligibleRewardTier))
-    } else {
-      deselectTier()
     }
-  }, [payAmountETH, rewardTiers, deselectTier, selectTier, selectedIndex])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedIndices, payAmountETH])
+
+  // sets pay input when selected nft's sum to greater than the current pay input amount
+  useEffect(() => {
```

Comment:
**question** useEffect feels like an antipattern here. Is there a way we can execute this logic explictly then calling `setSelectedIndices` instead?